File: benchmark/go/arrays.go
// arrays.go - Array Operations Benchmark for Go
// Tests dynamic array operations and iteration
package main

import (
	"fmt"
	"time"
)

func main() {
	size := 1000000

	start := time.Now()

	// Create array with 1,000,000 integers (0 to 999,999)
	arr := make([]int, size)
	for i := 0; i < size; i++ {
		arr[i] = i
	}

	// Sum all elements
	var sum1 int64 = 0
	for i := 0; i < size; i++ {
		sum1 += int64(arr[i])
	}

	// Reverse the array in-place
	for i := 0; i < size/2; i++ {
		arr[i], arr[size-1-i] = arr[size-1-i], arr[i]
	}

	// Sum again to verify
	var sum2 int64 = 0
	for i := 0; i < size; i++ {
		sum2 += int64(arr[i])
	}

	elapsed := time.Since(start).Milliseconds()

	fmt.Printf("Sum: %d\n", sum1)
	fmt.Printf("Reversed sum: %d\n", sum2)
	fmt.Printf("Array time: %dms\n", elapsed)
}

File: benchmark/rust/arrays.rs
// arrays.rs - Array Operations Benchmark for Rust
// Tests dynamic array operations and iteration

use std::time::Instant;

fn main() {
    let size = 1000000;

    let start = Instant::now();

    // Create array with 1,000,000 integers (0 to 999,999)
    let mut arr: Vec<i32> = (0..size).collect();

    // Sum all elements
    let sum1: i64 = arr.iter().map(|&x| x as i64).sum();

    // Reverse the array in-place
    arr.reverse();

    // Sum again to verify
    let sum2: i64 = arr.iter().map(|&x| x as i64).sum();

    let elapsed = start.elapsed().as_millis();

    println!("Sum: {}", sum1);
    println!("Reversed sum: {}", sum2);
    println!("Array time: {}ms", elapsed);
}
